dts/bindings/gpio/arduino-nano-header-r3.yaml

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2022 Joylab AG
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    GPIO pins exposed on Arduino Nano (R3) headers.
+
+    The Arduino Nano layout provides two headers on opposite edges of the board.
+
+    * A 15-pin header with mostly digital signals. The additional NRST (pin3)
+      and GND (pin 4) pins are not exposed by this binding.
+    * A 15-pin Analog Input and power supply header. This has analog input
+      signals labeled from A0 through A7, as well a digital signal D13. The
+      power supply and reset pins are not exposed by this binding.
+
+    This binding provides a nexus mapping for 22 pins where parent pins 0
+    through 7 correspond to A0 through A7, and parent pins 8 through 21
+    correspond to D0 through D13, as depicted below.
```

Comment:
Can you confirm that this number is consistent with the pin numbering used by pinMode() (https://www.arduino.cc/reference/en/language/functions/digital-io/pinmode/) for these boards?

Reply:
The pin numbering on the Nucleos is correct.
Doing a crosscheck using the Arduino Studio showed that on the Arduino Nano, D0 and D1 were interchanged. On the board itself those are marked as "RX0" and "TX1".
pinMode() takes as argument "0".."13" for the digital pins and "A0".."A7" for the analog pins.

Reply:
> pinMode() takes as argument "0".."13" for the digital pins and "A0".."A7" for the analog pins.

Yes, but A0 through A7 are just macros for pin numbers. I am asking if the numeric values you are assigning to pins here in this GPIO nexus node match the same numeric values that pinMode() expects.

Reply:
Adapted assignments, since nobody seemed to object.